ros2_control_demo_description/rrbot_description/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.5)
+project(rrbot_description)
+
+find_package(ament_cmake REQUIRED)
+
+install(
+  DIRECTORY config gazebo meshes ros2_control launch urdf
+  DESTINATION share/${PROJECT_NAME}
+)
+
+ament_environment_hooks("${CMAKE_CURRENT_SOURCE_DIR}/env-hooks/${PROJECT_NAME}.dsv.in")
```

Comment:
Would it be possible to elaborate on this specific file ? I am not familiar with it.
I found a related issue here:
https://github.com/ros-infrastructure/rosdep/issues/724
If I understand well it is to detect the rrbot_description package without resourcing the context ?

Compare with the issue, the package reference the source directory rather the install directory. Is this the intended behavior ?

Reply:
Remember that it initially took a very long time to start up Gazebo? That is because Gazebo's model path was not set to the custom install space. The ament environment hook takes care of this. You specify the source file, where the initial file can be found and `ament_environment_hooks` takes care of the rest during runtime.

Reply:
Nice trick.